src/measure_theory/vector_measure.lean

```diff
@@ -815,4 +815,85 @@ end
+
+open vector_measure
+
+open_locale measure_theory
+
+/-- The underlying function for `signed_measure.to_measure`. -/
+def to_measure' (s : signed_measure α) (i : set α) (hi : 0 ≤[i] s)
+  (j : set α) (hj : measurable_set j) : ℝ≥0∞ :=
+@coe ℝ≥0 ℝ≥0∞ _ ⟨s.restrict i j, le_trans (by simp) (hi j hj)⟩
+
+/-- Given a signed measure `s` and a positive measurable set `i`, `to_measure`
+provides the measure, mapping measurable sets `j` to `s (i ∩ j)`. -/
+def to_measure (s : signed_measure α) (i : set α) (hi₁ : measurable_set i) (hi₂ : 0 ≤[i] s) :
+  measure α :=
+measure.of_measurable (s.to_measure' i hi₂)
+  (by { simp_rw [to_measure', s.restrict_apply hi₁ measurable_set.empty,
+                 set.empty_inter i, s.empty], refl })
+  begin
+    intros f hf₁ hf₂,
+    have h₁ : ∀ n, measurable_set (i ∩ f n) := λ n, hi₁.inter (hf₁ n),
+    have h₂ : pairwise (disjoint on λ (n : ℕ), i ∩ f n),
+    { rintro n m hnm x ⟨⟨_, hx₁⟩, _, hx₂⟩,
+      exact hf₂ n m hnm ⟨hx₁, hx₂⟩ },
+    simp only [to_measure', s.restrict_apply hi₁ (measurable_set.Union hf₁),
+               set.inter_comm, set.inter_Union, s.of_disjoint_Union_nat h₁ h₂,
+               ennreal.some_eq_coe, id.def],
+    have h : ∀ n, 0 ≤ s (i ∩ f n),
+    { exact λ n, s.nonneg_of_zero_le_restrict
+          (s.zero_le_restrict_subset hi₁ (inter_subset_left _ _) hi₂) },
+    rw [nnreal.coe_tsum_of_nonneg h, ennreal.coe_tsum],
+    { refine tsum_congr (λ n, _),
+      simp_rw [s.restrict_apply hi₁ (hf₁ n), set.inter_comm] },
+    { exact (nnreal.summable_coe_of_nonneg h).2 (s.m_Union h₁ h₂).summable }
+  end
+
+variables (s : signed_measure α) {i j : set α}
+
+lemma to_measure_apply (hi : 0 ≤[i] s) (hi₁ : measurable_set i) (hj₁ : measurable_set j) :
+  s.to_measure i hi₁ hi j =
+  @coe ℝ≥0 ℝ≥0∞ _ ⟨s (i ∩ j), nonneg_of_zero_le_restrict s
+    (zero_le_restrict_subset s hi₁ (set.inter_subset_left _ _) hi)⟩ :=
+by { simp_rw [to_measure, measure.of_measurable_apply _ hj₁, to_measure',
+              s.restrict_apply hi₁ hj₁, set.inter_comm] }
+
+/-- `signed_measure.to_measure` is a finite measure. -/
+instance to_measure_finite (hi : 0 ≤[i] s) (hi₁ : measurable_set i) :
+  finite_measure (s.to_measure i hi₁ hi) :=
+{ measure_univ_lt_top :=
+  begin
+    rw [to_measure_apply s hi hi₁ measurable_set.univ],
+    exact ennreal.coe_lt_top,
+  end }
+
+lemma to_measure_to_signed_measure (hs : 0 ≤[univ] s) :
+  (s.to_measure univ measurable_set.univ hs).to_signed_measure = s :=
+begin
+  ext i hi,
+  simp [measure.to_signed_measure_apply_measurable hi, to_measure_apply _ _ _ hi],
+end
+
+end signed_measure
+
+namespace measure
+
+open vector_measure
+
+lemma to_signed_measure_to_measure (μ : measure α) [finite_measure μ] :
+  μ.to_signed_measure.to_measure univ measurable_set.univ
+    (restrict_le_restrict_of_subset_le _ _ (λ j hj₁ _,
+      by simp only [measure.to_signed_measure_apply_measurable hj₁, coe_zero, pi.zero_apply,
+                    ennreal.to_real_nonneg, vector_measure.coe_zero])) = μ :=
```

Comment:
This looks weird in a lemma statement. Would it be useful to have a lemma saying that `μ.to_signed_measure` is a positive measure on any subset, and use it here to formulate the statement?